src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md

```diff
@@ -0,0 +1,712 @@
+- `per-vu-iterations`
+- `constant-vus`
+- `ramping-vus`
+- `constant-arrival-rate`
+- `ramping-arrival-rate`
+- `externally-controlled`
+
+
+## Executors
+
+Executors are the workers of the k6 execution engine. Each one schedules VUs and
+iterations differently, and you'll choose one depending on the type of traffic you
+want to model to test your services.
+
+### Shared iterations
+
+A fixed number of iterations are "shared" between a number of VUs, and the test ends
+once all iterations are executed. This executor is equivalent to the global `vus` and
+`iterations` options.
+
+Note that iterations aren't fairly distributed with this executor, and a VU that
+executes faster will complete more iterations than others.
+
+#### Options
+
+| Option        | Type    | Description                                                                    | Default |
+|---------------|---------|--------------------------------------------------------------------------------|---------|
+| `vus`         | integer | Number of VUs to run concurrently.                                             | `1`     |
+| `iterations`  | integer | Total number of script iterations to execute across all VUs.                   | `1`     |
+| `maxDuration` | string  | Maximum test duration before it's forcibly stopped (excluding `gracefulStop`). | `"10m"` |
+
+#### When to use
+
+This executor is suitable when you want a specific amount of VUs to complete a fixed
+number of total iterations, and the amount of iterations per VU is not important.
+
+#### Examples
+
+- Execute 200 total iterations shared by 10 VUs with a maximum duration of 10 seconds:
+
+<div class="code-group" data-props='{"labels": [ "shared-iters.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  discardResponseBodies: true,
+  scenarios: {
+    contacts: {
+      executor: 'shared-iterations',
+      vus: 10,
+      iterations: 200,
+      maxDuration: '10s',
+    }
+  }
+};
+
+export default function() {
+  http.get('https://test.k6.io/contacts.php');
+}
+```
+
+</div>
+
+
+### Per VU iterations
+
+Each VU executes an exact number of iterations. The total number of completed
+iterations will be `vus * iterations`.
+
+#### Options
+
+| Option        | Type    | Description                                                                    | Default |
+|---------------|---------|--------------------------------------------------------------------------------|---------|
+| `vus`         | integer | Number of VUs to run concurrently.                                             | `1`     |
+| `iterations`  | integer | Number of script iterations to execute with each VU.                           | `1`     |
+| `maxDuration` | string  | Maximum test duration before it's forcibly stopped (excluding `gracefulStop`). | `"10m"` |
+
+#### When to use
+
+Use this executor if you need a specific amount of VUs to complete the same amount of
+iterations.
```

Comment:
```suggestion
Use this executor if you need a specific amount of VUs to complete the same amount of
iterations. This can be useful when you have fixed sets of test data that you want to 
partition between VUs.
```